### DISCUSSION TITLE
Converting Metal containing SMILES to mol

### DISCUSSION THREAD

**Discussion by Paul-Toth** [rdkit/rdkit#5280] [Q&A]
I have the following function that takes a dictionary of SMILES strings and converts them to RDKit mol objects.

    def smiles_dict_to_mol_list(smiles_dict):
        """smiles dict is a dictionary object containing molecule names
        as keys and smiles strings as values.
    
        The return value is a list of RDKit mol objects.
        """
    
        smiles_as_mol = []
        for mol_name, smiles_string in smiles_dict.items():
            try:
                mol = Chem.MolFromSmiles(smiles_string)
                mol.SetProp("_Name", mol_name)
                smiles_as_mol.append(mol)
            except:
                print("Error processing:", mol_name)
    
        return smiles_as_mol

This function normally works very well and I can convert any SMILES into a mol object, but when I try to convert the below structure I get an exception. I believe that this is because this compound, and others that also give exceptions, have a coordinated metal ion. In this case platinum. It is also possible that the SMILES string I am using is the problem, but I don't know how I would tell the difference.

[![enter image description here][1]][1]

N2W: `COc1cc(cc(c1O)OC)[C@@H]2c3cc4c(cc3[C@H]([C@@H]5[C@H]2C(=O)OC5)NC(=O)CC[C@@H]6C[NH2][Pt]([NH2]6)Cl)OCO4`

So my question is: **How can I change my RDKit code so I can work with compounds that have metal atoms?**

  [1]: https://i.stack.imgur.com/XZTx8.png

**Comment by wopozka** [comment]
The answer is very simple, your SMILES are wrong. Ligands bond to metals using coordination bond, which are not included in SMILES specification. If you use smiles for complexes then usually metal is treated as separate molecule, in this case it would be like: Cl[Pt]Cl.COc1cc(cc(OC)c1O)[C@H]1[C@H]2[C@H](COC2=O)[C@H](NC(=O)CC[C@@H](N)CN)c2cc3OCOc3cc12. What you can do is either to correct SMILES before reading them, or to read smiles without sanitization, correct given bonds inside mol (bonds betwen metal and other atom) and then sanitize mol.

**Reply by Paul-Toth** [reply]
Thank you @wopozka this is the answer that I found as well. I also [posted this question on StackOverflow](https://stackoverflow.com/questions/72205858/handling-smiles-with-metal-ions-in-rdkit) and was provided with the [following solution](https://www.rdkit.org/docs/Cookbook.html#organometallics-with-dative-bonds) from the RDKit cookbook on dative bonds.

This allows us to work with organometallic smiles and replace their covalent bonds with dative bonds when forming structures. Though you are correct that this method requires skipping sensitization while importing the structure.
